README.Rmd

```diff
@@ -38,9 +38,60 @@ remotes::install_github("inbo/territoria")
 
 ## Example
 
-This is a basic example which shows you how to solve a common problem:
+We start by simulating some observations.
+We need the for every observation their `x` and `y` coordinates in a projected coordinate system.
+`survey` is an integer id for every survey.
+A survey is a unique combination of an area and date.
+`status` is an integer indication the breeding status.
+A higher value assume more certainty about breeding.
+Set this to a constant value if you don't distinct between different certainties.
+In this example we use three classes: `1`, `2` and `3`.
 
-```{r example}
+```{r sim-data}
 library(territoria)
-## basic example code TODO
+set.seed(20210806)
+obs <- simulate_observations()
+names(obs)
+summary(obs$centroids)
+summary(obs$observations)
+obs <- obs$observations[obs$observations$observed, ]
+```
+
+Once we have a data.frame with the observations, we connect to a SQLite database and import the observations.
+This assigns every observation to its own cluster.
+
+```{r import-data}
+conn <- connect_db()
+import_observations(observations = obs, conn = conn, max_dist = 336)
+result <- get_cluster(conn = conn)
+nrow(result$observations) == nrow(result$cluster)
+```
+
+Next, we need to calculate the distance matrix.
+This is not the full distance matrix.
+We omit all irrelevant distances, e.g. between observations from the same survey or with a distance larger than twice the maximum cluster distance.
+
+```{r distance}
+distance_matrix(conn = conn, max_dist = 366)
+```
+
+Now we can start the clustering.
+The clustering takes into account all observations with a `status` greater than or equal to the set status.
+
+```{r cluster-3}
+cluster_observation(conn = conn, status = 3, max_dist = 336)
```

Comment:
It's not entirely clear now which type of clustering is performed (single, complete, average, centroid). Will different clustering possibilities be implemented later on?

Reply:
I've implemented complete linkage only. We can include other methods in the future.